### PR TITLE
Update TF dependencies to fix TF plan timing issue

### PR DESCRIPTION
**Pull request by umeshkumhar**
- Update frontend explicit depends_on which are not required
- Remove terraform provider that are not required in frontend
- update create_namespace to always true, because helm_release will take care if ns already exists and it wont attempt already exists error.

### CODE REVIEW

File: applications/rag/main.tf [hunk 1]
```diff
@@ -218,6 +214,5 @@ module "frontend" {
   url_domain_addr          = var.frontend_url_domain_addr
   url_domain_name          = var.frontend_url_domain_name
   members_allowlist        = var.frontend_members_allowlist
-
-  depends_on = [module.cloudsql, module.gcs, module.inference-server]
+  depends_on = [ module.gcs ]
```
Comment (hsachdevah) [inline applications/rag/main.tf:217]: We'll still need depend on module.cloudsql
rag-frontend deployment has cloud-sql-proxy which requires the CONNECTION_NAME.

**Review by hsachdevah — CHANGES_REQUESTED** [review]

File: applications/rag/main.tf (hunk 1 above)
Comment (hsachdevah) [reply applications/rag/main.tf:217]: It should be fine then.

**Review by hsachdevah — APPROVED** [review]

**Comment by artemvmin** [comment]
Could you please document what dependencies are being updated?

Additionally could you please document all the other bug fixes and changes being made? Generally, a person reading through the commit log and/or PR list shouldn't have to read the code to understand the changes that have been made. This also helps the reviewer understand what your intention is with the PR. For example, the following don't seem to be related to your PR title `Update TF dependencies to fix TF plan timing issue`:

- `ClientIP` -> `ClusterIP` typo
- Namespace is now always being created in some modules
  - Do we need this and does this have to be done in this PR? Let's reduce the scope to make sure we fix the timing issue completely.

**Pull request merged**